### PR TITLE
image: replace `GetStoreImage` with `ResolveReference` and  bump `c/image` to `373c52a9466f`

### PR DESCRIPTION
**Pull request by flouthoc**
`GetStoreImage` is deprecated after https://github.com/containers/image/pull/2056 use `ResolveReference` instead.

#### Does this PR introduce a user-facing change?

```release-note
* vendor: bump c/image to 373c52a9466f
* Remove `GetStoreImage` is deprecated use `ResolveReference` instead.
```

### CODE REVIEW

**Comment by flouthoc** [comment]
@containers/podman-maintainers PTAL

**Review by giuseppe — APPROVED** [review]
LGTM

File: libpod/storage.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func (r *storageService) CreateContainerStorage(ctx context.Context, systemConte
 		if err != nil {
 			return ContainerInfo{}, err
 		}
-		img, err := istorage.Transport.GetStoreImage(r.store, ref)
+		_, img, err := istorage.ResolveReference(ref)
```
Comment (vrothberg) [inline libpod/storage.go:83]: This function looks strange.  Especially the `imageName != ""` condition above.

**Review by vrothberg — COMMENTED** [review]
Code LGTM

I dropped a comment in the function.  The condition looks strange to me.  Line 73 checks for `imageName != ""` and line 101 has a `imageName == ""` under this very condition.

@flouthoc, if you have time, could you track down what this code is supposed to do?

File: libpod/storage.go (hunk 1 above)
Comment (Luap99) [reply libpod/storage.go:83]: tracked in https://github.com/containers/podman/issues/20411

**Pull request merged**